src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java

```diff
@@ -15,139 +20,264 @@
         }
-
-        throw new ConfigException("Malformed topics to kusto ingestion props mappings");
     }
 
     public TopicIngestionProperties getIngestionProps(String topic) {
         return topicsToIngestionProps.get(topic);
     }
 
+    void validateTableMappings(KustoSinkConfig config) {
+        List<String> databaseTableErrorList = new ArrayList<>();
+        List<String> accessErrorList = new ArrayList<>();
+        try {
+            Client engineClient = createKustoEngineClient(config);
+            if (config.getTopicToTableMapping() != null) {
+                JSONArray mappings = new JSONArray(config.getTopicToTableMapping());
+                for (int i = 0; i < mappings.length(); i++) {
+                    JSONObject mapping = mappings.getJSONObject(i);
+                    validateTableAccess(config, engineClient, mapping, databaseTableErrorList, accessErrorList);
+                }
+            }
+            String tableAccessErrorMessage = "";
+
+            if(!databaseTableErrorList.isEmpty())
+            {
+                tableAccessErrorMessage = "\n\nUnable to access the following database:table\n" +
+                        String.join("\n",databaseTableErrorList);
+            }
+            if(!accessErrorList.isEmpty())
+            {
+                tableAccessErrorMessage = tableAccessErrorMessage + "\n\nUser does not have appropriate permissions " +
+                        "to sink data into the Kusto database:table combination(s). " +
+                        "Verify your Kusto principals and roles before proceeding for the following: \n " +
+                        String.join("\n",accessErrorList);
+            }
+
+            if(!tableAccessErrorMessage.isEmpty()) {
+                throw new ConnectException(tableAccessErrorMessage);
+            }
+        } catch (JSONException e) {
+            throw new ConnectException("Failed to parse ``kusto.tables.topics.mapping`` configuration.", e);
+        }
+    }
+
+    /**
+     *  This function validates whether the user has the read and write access to the intended table
+     *  before starting to sink records into ADX.
+     * @param config KustoSinkConfig class as defined by the user.
+     * @param engineClient Client connection to run queries.
+     * @param mapping JSON Object containing a Table mapping.
+     */
+    private static void validateTableAccess(KustoSinkConfig config, Client engineClient, JSONObject mapping, List<String> databaseTableErrorList, List<String> accessErrorList) throws JSONException {
+        String getPrincipalsQuery = ".show table %s principals";
+        String database = mapping.getString("db");
+        String table = mapping.getString("table");
+        try {
+            Results rs = engineClient.execute(database, String.format(getPrincipalsQuery, table));
+            String authenticateWith;
+            if (config.getAuthAppid() != null) {
```

Comment:
I think a better way would be to use '.show principal roles' command.

Reply:
The query would be `.show table <tableName> principal roles` and simply check it has 'ingestor' role
Its better also as table ingestor can't use `.show table %s principals`

Reply:
Thanks for sharing this, it works like a charm.
Just wanted to confirm that it returns only a single role with which it's logged-in with.
This is something that we observed while having multiple roles for the table, I might be wrong.